apps/angular/src/app/features/auth/components/registration-form/registration-form.component.ts

```diff
@@ -0,0 +1,67 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+import { UserService } from '@js-camp/angular/core/services/user.service';
+import { catchValidationData } from '@js-camp/angular/core/utils/rxjs/catch-validation-error';
+import { Destroyable, takeUntilDestroy } from '@js-camp/angular/core/utils/rxjs/destroyable';
+import { toggleExecutionState } from '@js-camp/angular/core/utils/rxjs/toggle-execution-state';
+import { AppValidators } from '@js-camp/angular/core/utils/validators';
+import { BehaviorSubject } from 'rxjs';
+
+/** Registration form. */
+@Destroyable()
+@Component({
+  selector: 'camp-registration-form',
+  templateUrl: './registration-form.component.html',
+  styleUrls: ['./registration-form.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class RegistrationFormComponent {
+  /** Is Loading. */
+  public isLoading$ = new BehaviorSubject<boolean>(false);
+
+  /** Registration form. */
+  public registrationForm = this.formBuilder.group({
+    email: this.formBuilder.nonNullable.control('', [
+      Validators.email,
+      Validators.required,
+    ]),
+
+    firstName: this.formBuilder.nonNullable.control('', [Validators.required]),
+
+    lastName: this.formBuilder.nonNullable.control('', [Validators.required]),
+
+    password: this.formBuilder.nonNullable.control('', [Validators.required]),
+
+    passwordConfirm: this.formBuilder.nonNullable.control('', [Validators.required, AppValidators.matchControl('password')]),
+  });
```

Comment:
```suggestion
    return this.formBuilder.nonNullable.group({
      email: ['', [Validators.required, Validators.email]],
      firstName: ['', [Validators.required]],
      lastName: ['', [Validators.required]],
      password: ['', [Validators.required, Validators.pattern(passwordPattern)]],
      passwordConfirm: ['', [Validators.required, AppValidators.matchControl('password')]],
  });
```
You could do it like this, it's less verbose this way